### PR TITLE
Feature/basic ui

### PR DESCRIPTION
**Pull request by RobertWSaunders**
- adds detailed readme
- npm scripts clean up
- webpack config w/ code splitting, sass, fonts, auto injecting
- dotenv setup (removed config in favour of dotenv, @zackharley read comment below)
- modified eslintrc file with specified rules, no longer js file, see packages, ide integration
- adds sass linting yaml file
- add client file structure

### CODE REVIEW

**Comment by RobertWSaunders** [comment]
@zackharley I removed the config package you added, the reasoning was because the config folder had to live in the root directory by default. It could be overridden by an environment variable so I did so to put in the `server` folder but then realized it only had the port in it, so as per the README, i have implemented logic in `server/index.js` that makes the default port 3000 but a environment variable called PORT can override. I suggest we use `dotenv` for environment variables and then make a `config.json` for other things. I have laid out some potential other environment variables we may want to set up in the README.

**Comment by RobertWSaunders** [comment]
@zackharley when this is approved I'll move onto creating a basic routing system and store, from the the team can make components to flush out the UI.

File: app.yaml [hunk 1]
```diff
@@ -1,9 +1,9 @@
 runtime: nodejs
 env: flex
 automatic_scaling:
-  max_num_instances: 1
```
Comment (zackharley) [inline app.yaml]: Why were these changes made?

File: package.json [hunk 2]
```diff
@@ -1,22 +1,49 @@
 {
   "name": "hacker-dashboard",
-  "version": "0.0.0",
-  "description": "A dashboard for QHacks participants",
-  "author": "",
+  "version": "1.0.0",
```
Comment (zackharley) [inline package.json]: I don't think we've released `v1.0.0` so I don't think this is correct. I think `v1.0.0 will come when we are ready to launch.

File: package.json [hunk 3]
```diff
@@ -1,22 +1,49 @@
...
+  "description": "Dashboard for participants, judges, mentors and partners of QHacks 2018.",
+  "author": "Robert Saunders, Zack Harley, Cory Crowley & Avi Ihejirika",
   "scripts": {
+    "dev": "NODE_ENV=development nodemon server",
+    "start": "NODE_ENV=production node server",
+		"ui": "webpack-dev-server",
```
Comment (zackharley) [inline package.json]: I don't think should be indented like this

File: package.json [hunk 4]
```diff
@@ -1,22 +1,49 @@
...
+    "build": "NODE_ENV=production npm run clean && webpack -p",
+    "test": "eslint server/**.js client/**.js",
     "deploy": "gcloud app deploy",
-    "lint": "eslint server/**.js",
-    "start": "node server"
+    "clean": "rimraf ./client/bundle"
```
Comment (zackharley) [inline package.json]: I don't see `rimraf` listed as a dependency anywhere. Why are we using this over `rm -rf`?

File: server/index.js [hunk 5]
```diff
@@ -1,15 +1,15 @@
...
 const bodyParser = require('body-parser');
-const config = require('config');
 const express = require('express');
 const routes = require('./routes/index.route');
 const path = require('path');
 const winston = require('winston');
 
 const app = express();
-const port = config.get('server.port');
+let port = process.env.PORT || 3000;
```
Comment (zackharley) [inline server/index.js]: Why `let`?

File: webpack.config.js [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+ */
+const VENDOR_LIBS = [
+	'react',
+	'react-redux',
+	'redux',
+	'redux-form'
+];
+
+/**
+ * Webpack configuation object.
```
Comment (zackharley) [inline webpack.config.js]: I appreciate the want to document our code, but this seems like its pretty self-explanatory.

**Comment by zackharley** [comment]
@RobertWSaunders I'm still not clear on why `dotenv` over `config`

File: app.yaml (hunk 1 above)
Comment (RobertWSaunders) [reply app.yaml]: my apologies, editor automatically converts spaces to tabs, have addressed as yaml files should use spaces

Comment (zackharley) [reply app.yaml]: No worries, was just curious about why the changes were made

File: package.json (hunk 2 above)
Comment (RobertWSaunders) [reply package.json]: Fair, 0.0.0 felt weird though, minor caveat, not that it's even going on a registry.

Comment (zackharley) [reply package.json]: Yeah, I figured `1.0.0` would be the version we release with apps (Nov.13) or when we're ready to test with our exec.

File: package.json (hunk 4 above)
Comment (RobertWSaunders) [reply package.json]: `rimraf` will perform same operation on all OS's, provides better code, does need to be added as dependancy, will fix, have a global currently

File: server/index.js (hunk 5 above)
Comment (RobertWSaunders) [reply server/index.js]: can be `const`, will fix

File: webpack.config.js (hunk 6 above)
Comment (RobertWSaunders) [reply webpack.config.js]: I like being fairly explicit, but can remove.

**Comment by RobertWSaunders** [comment]
@zackharley In my opinion it provides more benefits and achieves the same thing with less effort.
1. `.env` files are user specific, better security, better local dev flexibilty
2. build does not depend on `.env` file, `config` does
3. can share between front and back end in a cleaner way
4. leverages environment variable config we already use for other things
5. follows twelve factor app style, strict separation between config and code

**Comment by zackharley** [comment]
@RobertWSaunders Alright, `.env` works for me

File: package.json (hunk 3 above)
Comment (RobertWSaunders) [reply package.json]: oops, rogue tab in there

File: app.yaml [hunk 7]
```diff
@@ -1,9 +1,9 @@
 runtime: nodejs
 env: flex
 automatic_scaling:
-	max_num_instances: 1
-	cpu_utilization:
-		target_utilization: 0.5
+  max_num_instances: 1
+  cpu_utilization:
+  target_utilization: 0.5
```
Comment (zackharley) [inline app.yaml]: Needs to be indented once more

File: app-stage.yaml [hunk 8]
```diff
@@ -1,9 +1,9 @@
 runtime: nodejs
 env: flex
 automatic_scaling:
-	max_num_instances: 1
-	cpu_utilization:
-		target_utilization: 0.5
+  max_num_instances: 1
+  cpu_utilization:
+  target_utilization: 0.5
```
Comment (zackharley) [inline app-stage.yaml]: Needs to be indented once more

**Pull request merged**